src/tests/resistance/fuzzing.rs

```diff
@@ -21,14 +22,19 @@ use crate::{
         node::{Action, Node},
     },
 };
+use std::sync::Arc;
 
 use rand::{
     distributions::Standard,
     prelude::{Rng, SeedableRng, SliceRandom},
     thread_rng,
 };
 use rand_chacha::ChaCha8Rng;
-use tokio::{io::AsyncWriteExt, net::TcpStream};
+use tokio::{
+    io::AsyncWriteExt,
+    net::{TcpListener, TcpStream},
+    sync::Mutex,
```

Comment:
I'd use the parking lot variant of `Mutex` as we don't need to hold the lock over an `await`. I'd also use an `RwLock` instead since the rng is `Sync` + `Send`. https://docs.rs/lock_api/0.4.4/lock_api/struct.Mutex.html

Reply:
I wasn't aware there were differences - makes sense.